client/routes/domain/workflow-list.vue

```diff
@@ -428,12 +443,12 @@ export default {
       return query;
     },
     toggleFilter() {
-      if (this.filterMode === 'advanced') {
-        this.filterMode = 'basic';
-        this.$route.query.queryString = '';
-      } else {
-        this.filterMode = 'advanced';
-      }
+      const { query } = this.$route;
+
+      this.clearState();
+      const filterMode = this.filterMode === 'advanced' ? 'basic' : 'advanced';
+
+      this.$router.replace({ query: { ...query, filterMode } });
```

Comment:
When toggling between advanced or basic, it will keep track of this value in a query param so on refresh of the page it will stay on the page you were looking at previously.